paderbox/transform/module_fbank.py

```diff
@@ -85,166 +135,342 @@ def inverse(self, x):
+    >>> number_of_filters = 10
+    >>> frequency = mel2hz(np.linspace(hz2mel(lowest_frequency), hz2mel(highest_frequency), number_of_filters+2))
+    >>> frequency
+    array([   0.        ,  180.21928115,  406.83711843,  691.7991039 ,
+           1050.12629534, 1500.70701371, 2067.29249375, 2779.74887082,
+           3675.63149949, 4802.16459006, 6218.73051459, 8000.        ])
+    >>> f_warped = hz_warping(frequency, warp_factor=1.1, boundary_frequency_ratio=.6, highest_frequency=highest_frequency)
+    >>> f_warped
+    array([   0.        ,  198.24120926,  447.52083027,  760.97901429,
+           1155.13892487, 1650.77771509, 2274.02174313, 3057.7237579 ,
+           4043.19464944, 5185.90483925, 6432.48285284, 8000.        ])
+    >>> f_warped = hz_warping(frequency, warp_factor=[.9, 1.1], boundary_frequency_ratio=.6, highest_frequency=highest_frequency)
+    >>> f_warped
+    array([[   0.        ,  162.19735303,  366.15340659,  622.61919351,
+             945.11366581, 1350.63631234, 1860.56324438, 2501.77398374,
+            3308.06834954, 4322.48927857, 5951.54009178, 8000.        ],
+           [   0.        ,  198.24120926,  447.52083027,  760.97901429,
+            1155.13892487, 1650.77771509, 2274.02174313, 3057.7237579 ,
+            4043.19464944, 5185.90483925, 6432.48285284, 8000.        ]])
+    >>> f_warped/frequency
+    array([[       nan, 0.9       , 0.9       , 0.9       , 0.9       ,
+            0.9       , 0.9       , 0.9       , 0.9       , 0.90011269,
+            0.95703457, 1.        ],
+           [       nan, 1.1       , 1.1       , 1.1       , 1.1       ,
+            1.1       , 1.1       , 1.1       , 1.1       , 1.07990985,
+            1.03437234, 1.        ]])
+    >>> hz_warping(frequency, warp_factor=[[.9], [1.1]], boundary_frequency_ratio=.75, highest_frequency=highest_frequency).shape
+    (2, 1, 12)
     """
-    return f * fft_length / sample_rate
+    assert (np.max(frequency) - highest_frequency) < 1e-6, (
+        np.max(frequency), highest_frequency
+    )
+    warp_factor = np.array(warp_factor)
+    boundary_frequency = np.array(boundary_frequency_ratio) * highest_frequency
+    breakpoints = boundary_frequency * np.minimum(warp_factor, 1) / warp_factor
+
+    if breakpoints.ndim == 0:
+        breakpoints = np.array(breakpoints)
+    breakpoints[
+        (breakpoints > highest_frequency)
+        + ((warp_factor * breakpoints) > highest_frequency)
+    ] = highest_frequency
+    bp_value = warp_factor * breakpoints
+
+    frequency, breakpoints, bp_value = np.broadcast_arrays(
+        frequency, breakpoints[..., None], bp_value[..., None]
+    )
+    f_warped = warp_factor[..., None] * frequency
+    idx = frequency > breakpoints
+    f_warped[idx] = (
+            highest_frequency
+            + (
+                    (frequency[idx] - highest_frequency)
+                    * (highest_frequency - bp_value[idx]) / (highest_frequency - breakpoints[idx])
+        )
+    )
```

Comment:
If `boundary_frequency_ratio` happens to be greater as 1 (e.g., wrong sampling function), this part throws a divide-by-zero error:
```python
paderbox/paderbox/transform/module_fbank.py:338: RuntimeWarning: divide by zero encountered in true_divide
  * (highest_frequency - bp_value[idx]) / (highest_frequency - breakpoints[idx])
```
Since the docstring says that `boundary_frequency_ratio` must be in [0, 1], maybe you could add an assert for that.